File: 009_palindrome_number/Solution.java
public class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;

        int result = 0;
        int y = x;

        try {
            while (y != 0) {
                result = result * 10 + y % 10;
                y /= 10;
            }
        } catch (Exception e) {
            return false;
        }

        return x == result;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.isPalindrome(12321));
    }
}

File: 009_palindrome_number/Solution.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0: return False

        result = 0
        y = x
        max_int = 2 ** 31 - 1

        while (y != 0):
            result = result * 10 + y % 10
            if result > max_int: return False
            y //= 10

        return x == result


if __name__ == '__main__':
    solution = Solution()
    print(solution.isPalindrome(12321))
